pkg/collector/containerdlogs_collector.go

```diff
@@ -0,0 +1,106 @@
+package collector
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/aks-periscope/pkg/interfaces"
+	"github.com/Azure/aks-periscope/pkg/utils"
+)
+
+// ContainerDLogsCollector defines a ContainerDLogs Collector struct
+type ContainerDLogsCollector struct {
+	BaseCollector
+}
+
+// ContainersList defines configuration for list of containers and their details
+type ContainersList struct {
+	ContainersList []ContainerInformation `json:"containers"`
+}
+
+// ContainerInformation defines configuration of information available for each container
+type ContainerInformation struct {
+	ID     string            `json:"id"`
+	Labels map[string]string `json:"labels"`
+}
+
+var _ interfaces.Collector = &ContainerDLogsCollector{}
+
+// NewContainerDLogsCollector is a constructor
+func NewContainerDLogsCollector(exporter interfaces.Exporter) *ContainerDLogsCollector {
+	return &ContainerDLogsCollector{
+		BaseCollector: BaseCollector{
+			collectorType: ContainerDLogs,
+			exporter:      exporter,
+		},
+	}
+}
+
+// Collect implements the interface method
+func (collector *ContainerDLogsCollector) Collect() error {
+	containerLogs := strings.Fields(os.Getenv("DIAGNOSTIC_CONTAINERLOGS_LIST"))
+	rootPath, err := utils.CreateCollectorDir(collector.GetName())
+	if err != nil {
+		return err
+	}
+
+	output, err := utils.RunCommandOnHost("crictl", "ps", "-a", "-o", "json")
+	if err != nil {
+		return err
+	}
+
+	var containers ContainersList
+	if err = json.Unmarshal([]byte(output), &containers); err != nil {
+		log.Fatalf("unable to read containers json file: %v", err)
+	}
+
+	for _, containerLog := range containerLogs {
+		containerLogParts := strings.Split(containerLog, "/")
+		targetNameSpace := containerLogParts[0]
+		containerNames := map[string]string{}
+
+		for _, container := range containers.ContainersList {
+			containerID := container.ID
+			containerNameSpace, ok := container.Labels["io.kubernetes.pod.namespace"]
+			if !ok {
+				log.Printf("Unable to retrieve namespace of container: %+v", containerID)
+				continue
+			}
+
+			containerName, ok := container.Labels["io.kubernetes.container.name"]

```

Comment:
💡 How is this `containerName` used? i.e. see the line below where code have `!ok` the logic still continues does that means `containerName` will always be there? but  `!ok` suggest otherwise.

Reply:
Here is an assumption that not all containers are managed by Kubernetes, so when we try to retrieve the container name using this label, the key "io.Kubernetes.container.name" may not exist. In this case, we just skip collecting logs of this container, thus we use `continue` to skip this cycle of for loop.